src/cache/redis.rs

```diff
@@ -2,37 +2,95 @@ use crate::twin::Twin;
 
 use super::Cache;
 
-use anyhow::Result;
+use anyhow::{Context, Result};
 use async_trait::async_trait;
-use bb8_redis::{bb8::Pool, RedisConnectionManager};
-use serde::{Deserialize, Serialize};
+use bb8_redis::{
+    bb8::{Pool, PooledConnection},
+    redis::{cmd, FromRedisValue, ToRedisArgs},
+    RedisConnectionManager,
+};
+use serde::{de::DeserializeOwned, Serialize, Serializer};
 
 //
 // how_to_init
 // let manager = RedisConnectionManager::new(url)?;
 // let pool = Pool::builder().build(manager).await?;
+// RedisCache::new(pool, "twin", 10); //10 seconds
 //
 
 #[derive(Clone)]
-pub struct RedisCache {
+pub struct RedisCache<P>
+where
+    P: Into<String>,
+{
     pool: Pool<RedisConnectionManager>,
+    prefix: P,
```

Comment:
prefix imho, should be `String` because this is need to be owned always by the RedisCace. The idea of using `Into<String>` is only in function parameters. but not on the type itself.

Let me explain

```rust
struct MyStruct<T>;

impl<T> MyStruct<T> where T: Into<String> {
   fn function(&self, t: T) {
   }
}
```

The problem here now is that once you created your type
```rust
let s: MyStruct<String> = MyStruct;

// you can only call it like
s.function(String::from("hello"));
// but not like
s.function("hello"); 
```

BUT if we do it differently like this

```rust
struct MyStruct;

impl MyStruct  {
   fn function<T: Into<String>(&self, t: T) {
       let s = t.into(); // this returns a string
   }
}
```

The problem here now is that once you created your type
```rust
let s = MyStruct;

// you can call it like
s.function(String::from("hello"));
// this will also work
s.function("hello"); 
```

because the compiler will have 2 version of the function method that accepts both String and &str. (and anything that implements the Into<String> trait.

that what I meant from the guide lines.

Hence this code instead need to be like
```rust
struct RedisCache{
 prefix: String
}

impl RedisCache {
   fn new<S: Into<String>(s: prefix) {
       RedisCache{prefix: s.into()}
   }
}

// then u can create an instance like
let cache = RedisCache::new("hello");
// or 
let cache = RedisCache::new(String::from("hello")); 
```